### PR TITLE
Implements proposed max-template-depth rule

### PR DESCRIPTION
**Pull request by kevsommer**
This PR addresses issue https://github.com/vuejs/eslint-plugin-vue/issues/2524.

I believe it successfully implements the rule, that the test cases are exhaustive enough. 

Hoping for feedback! :)

### CODE REVIEW

File: docs/rules/max-template-depth.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        </div>
+      </nested-component>
+    </child-component>
+  </main-container>
+</template>
+```
+
+</eslint-code-block>
+
+### `{ maxDepth: 5 }`
```
Comment (FloEdelmann) [inline docs/rules/max-template-depth.md:49]: Either remove this heading (so both good and bad examples use `3`), or adjust the `3` to `5` in the line below.

File: docs/rules/max-template-depth.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+This rule enforces a maximum depth of the template in a Vue SFC, in order to aid in maintainability and reduce complexity.
+
+## :wrench: Options
+
+This rule takes an object, where you can specify the maximum depth allowed in a Vue SFC template block.
+There is one property that can be specified for the object.
+
+- `maxDepth` ... Specify the maximum template depth `template` block.
+
+## `{ maxDepth: 3 }`
```
Comment (FloEdelmann) [inline docs/rules/max-template-depth.md:25]: This should be a sub-heading of the "Options" section. That's also the reason why `npm run update` and thus the preview deployment on Netlify is currently failing.

```suggestion
### `{ maxDepth: 3 }`
```

File: lib/rules/max-template-depth.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            minimum: 1
+          }
+        },
+        additionalProperties: false,
+        minProperties: 1
+      }
+    ],
+    messages: {
+      templateTooDeep:
+        'Template is too deep (depth of {{depth}}). Maximum allowed is {{limit}}.'
```
Comment (FloEdelmann) [inline lib/rules/max-template-depth.js:31]:
```suggestion
        'Element is nested too deeply (depth of {{depth}}, maximum allowed is {{limit}}).'
```

**Review by FloEdelmann — APPROVED** [review]
Looks good to me :slightly_smiling_face: 
Thanks for your contribution!

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Review by SamyDgz — APPROVED** [review]

**Pull request merged**